### DISCUSSION TITLE
How do I create a Composer patch for OpenMage files?

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#4087] [Questions & Answers]
OpenMage has inherited from Magento the feature to change the core files by copying them to /app/code/local. Is it possible to use Composer with patches to change some core files or add some PR's from other repositories?

In the vendor directory there is the **openmage** subdirectory. I can create another subdirectory **magento-lts**. In composer.json I can add like for shardh/zf1-future a new section for openmage/magento-lts with the all changes I need in extra > patches.

Running composer update nothing happens. Am I doing something wrong? What shall I do next?

**Comment by fballiano** [comment]
did you try removing the vendor directory and redo a composer install?

**Reply by addison74** [reply]
I did not remove the vendor directory. I can create patches for any package installed by Composer, except the files related to OpenMage. For example, I need some PRs which are only available in v21-beta but I don't know how to integrate them. Composer allows me to revert to initial stage. I don't want to use git command because this installation will go in production.

If someone managed to do this task please share a tutorial.

**Comment by SuSiSonnenschein111** [comment]
From my knowledge "vendor patches" only work when OM is installed via composer - but not for git clone installs.

Untested idea ... apply the patches via composers post-install/update hook.
